### ISSUE TITLE
Gzip::IsolatedInitialize not propagate to its base class

### ISSUE THREAD

**Issue by asbai** [weidai11/cryptopp#660]
`Deflator::IsolatedInitialize` is not called in the `Gzip::IsolatedInitialize` method. It should be called, otherwise the data is broken when we called the `Initialize` method and try to begin to compress another chunk of data.

You can fix this bug by:

```
void Gzip::IsolatedInitialize(const NameValuePairs &parameters)
{
	Deflator::IsolatedInitialize(parameters);  // <--- Add this line of code

	ConstByteArrayParameter v;
	if (parameters.GetValue(Name::FileName(), v))
		m_filename.assign(reinterpret_cast<const char*>(v.begin()), v.size());
	if (parameters.GetValue(Name::Comment(), v))
		m_comment.assign(reinterpret_cast<const char*>(v.begin()), v.size());
	m_filetime = parameters.GetIntValueWithDefault(Name::FileTime(), 0);
}
```

**Comment by noloader** [comment]
Thanks @asbai.

I'm not sure why that was not added when we added the filename, filetime and comment. Gzip has been around a long time. The additional fields were added at [Commit d901ecd9a4de](https://github.com/weidai11/cryptopp/commit/d901ecd9a4de) for Crypto++ 6.0.

@mouse07410, @MarcelRaad, does this look OK to you?

**Comment by noloader** [comment]
Thanks @asbai.

I'm not sure why that was not added when we added the filename, filetime and comment. Gzip has been around a long time. The additional fields were added at [Commit d901ecd9a4de](https://github.com/weidai11/cryptopp/commit/d901ecd9a4de) for Crypto++ 6.0.

I don't think we have a test case that calls `IsolatedInitialize` with different filenames. I'm not sure if things will work as expected.

@mouse07410, @MarcelRaad, does this look OK to you?
